components/builder-worker/src/runner/mod.rs

```diff
@@ -331,10 +332,15 @@ impl Runner {
         Ok(archive)
     }
 
-    fn do_export(&mut self, tx: &mpsc::Sender<Job>, mut streamer: &mut JobStreamer) -> Result<()> {
+    fn do_export(
+        &mut self,
+        tx: &mpsc::Sender<Job>,
+        mut archive: &mut PackageArchive,
```

Comment:
Ok so, bear with me here because I'm NOT fully versed in the build worker code and thus we'll definitely want some @chefsalim feedback here as well but I also had some questions.

It looks like you're hauling this archive through almost explicitly for the sake of grabbing the package target for branching logic in the export function. If that's the case I don't know that it's actually totally necessary.

I _think_ you can probably get the `PackageTarget` from the archive generated by `do_build` without passing in the extra argument as do_build is just returning `Result<PackageArchive>`

What i think that means is changing the function signature in `do_export` back to the way it was and then at line 410 you've actually already got the archive you need which you should be able to get the target from with a:
`let target = archive.target()?;` and then maybe you could change the `export` function sig to something like:
```
fn export(&mut self, archive: PackageTarget, streamer: &mut JobStreamer) -> Result<()>
```
Then in your export function you can get rid of line 515.

I think that might make this a little more lightweight. If you've tried all this/i'm missing something feel free to tell me i'm an idiot :joy:

Reply:
Right, yeah I think we can just make this change in the export function, without making any signature changes at all.  You should be able to check `self.workspace.job.target` and skip doing the export if it is Linux2.  It may be good to also emit a debug! message when skipping due to target.

Reply:
My first pass was similar to that ( I had the logic wrapped around do_export, rather than pass the archive in),  but decided to go this route because the [following line](https://github.com/habitat-sh/builder/blob/master/components/builder-worker/src/runner/mod.rs#L509-L510 ) could also use `archive.is_a_service()` if we address fnichols comment about the `is_a_service` method on PackageArchive and I _think_ that is probably more correct than "last built".  I'm probably overthinking things though since we don't need to address that as part of this change.   

I think you're right right though, and should make the change need today and not the one we might need at some point in the future.

Reply:
Yeah the is_a_service thing is a bigger discussion...